roles/ocp-route-redirect/tasks/redirect.yml

```diff
@@ -0,0 +1,19 @@
+---
+
+- k8s_facts:
+    api_version: route.openshift.io/v1
+    kind: Route
+    name: "{{ route_redirect_name }}"
+    namespace: "{{ route_redirect_namespace }}"
+  register: route
+
+- set_fact:
+    route: "http://{{ (route.resources | first).spec.host }}"
+
+- uri:
+    url: "{{ route }}"
+    validate_certs: "{{ route_redirect_validate_certs | default(false) }}"
+  register: redirect_url
```

Comment:
My suggestion would be to put this in a loop and keep going till it no longer redirects (or maybe quit after ~X number of redirects if it didn't successfully find a valid target)

Reply:
I actually ran into some troubles here -- so maybe you can help point me in the right direction. This returns everything in redirect_url, so I can tell if the route was redirected by checking {{ redirect_url.redirected }} and get the url from {{ redirect_url.url }}. My problem with looping was with getting the updated {{ redirect_url.url }} back to the uri module. I tried this with something like the below without any luck:

```
- uri:
     url: "{{ redirect_url.url }}"
     validate_certs: false
   register: redirect_url
   until: not redirect_url.redirected
```

I've attempted this with loop, with_items, etc. -- but the problem seems to be getting the updated url back to this module. Happy to discuss some of the failures offline. But if you have any thoughts here I'm happy to give it a shot.